bridge_adaptivity/module/models.py

```diff
@@ -110,6 +113,86 @@ def get_absolute_url(self):
         return reverse('module:collection-list')
 
 
+class Engine(models.Model):
+    """Defines engine settings."""
+
+    DEFAULT_ENGINE_NAME = 'mock'
+
+    name = models.CharField(default=DEFAULT_ENGINE_NAME, max_length=255)
+    host = models.URLField(blank=True, null=True)
+    token = models.CharField(max_length=255, blank=True, null=True)
+
+    _engines_cache = {}
+
+    class Meta:
+        unique_together = (('host', 'token'),)
+
+    def __str__(self):
+        return "Engine: {}".format(self.name)
+
+    @classmethod
+    def get_default_engine(cls):
+        if not cls.objects.count():
+            Engine.objects.create()
+            return cls.objects.get()
+        else:
+            return cls.objects.get(name=cls.DEFAULT_ENGINE_NAME)
+
+    def get_absolute_url(self):
+        return ''
+
+    def make_engine(self):
+        try:
+            engine_module = importlib.import_module('module.engines.engine_{}'.format(self.name.lower()))
+        except ImportError:
+            engine_module = importlib.import_module('module.engines.engine_mock')
+        engine_template = 'Engine{}'
+        driver = getattr(
```

Comment:
ditto, with the fixture of the engines `name` should be equal to the driver's name in the engine's module

Reply:
this is fallback in case if misconfigured.

Reply:
ok, we will return to the engine in next PRs